addons/controllers/csi/vspherecsiconfig_controller.go

```diff
@@ -160,10 +159,12 @@ func (r *VSphereCSIConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	// deep copy VSphereCSIConfig to avoid issues if in the future other controllers where interacting with the same copy
 	vcsiConfig = vcsiConfig.DeepCopy()
-
-	cluster, err := r.getOwnerCluster(ctx, vcsiConfig)
+	cluster, err := cutil.GetOwnerCluster(ctx, r.Client, vcsiConfig, req.Namespace, constants.CSIDefaultRefName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
 	if cluster == nil {
```

Comment:
What about the case where cluster is not nil and apierrors.IsNotFound ?

Reply:
added logic to handle